drivers/i2s/i2s_litex.c

```diff
@@ -0,0 +1,617 @@
+/*
+ * Copyright (c) 2020 Antmicro <www.antmicro.com>
+ * 
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <drivers/i2s.h>
+#include <sys/byteorder.h>
+#include <soc.h>
+#include <sys/util.h>
+#include <assert.h>
+#include "i2s_litex.h"
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(i2s_litex);
+#define DEV_CFG(dev) ((struct i2s_litex_cfg *const)(dev)->config_info)
+#define DEV_DATA(dev) ((struct i2s_litex_data *const)(dev)->driver_data)
+
+#define MODULO_INC(val, max)                                                   \
+	{                                                                      \
+		val = (++val < max) ? val : 0;                                 \
+	}
+
+/**
+ * @brief Enable i2s device
+ *
+ * @param reg base register of device 
+ *
+ * @return N/A
+ */
+static void i2s_enable(int reg)
```

Comment:
Consider using uintptr_t instead of int to make it clearer this is an address.

Reply:
Changed here and in simillar functions